src/app/Shared/Services/Login.service.tsx

```diff
@@ -188,23 +188,64 @@ export class LoginService {
     return this.logout.asObservable();
   }
 
-  setLoggedOut(): void {
-    this.removeCacheItem(this.USER_KEY);
-    this.removeCacheItem(this.TOKEN_KEY);
-    this.token.next('');
-    this.username.next('');
-    this.logout.next();
-    this.sessionState.next(SessionState.NO_USER_SESSION);
-
-    // TODO implement a logout call that will delete the access_token
-    // cached in the oauth server, thus redirecting users to the login page
-    this.queryAuthMethod();
+  setLoggedOut(): Observable<boolean> {
+    return combineLatest([this.getToken(), this.getAuthMethod()]).pipe(
+      first(),
+      concatMap(parts => {
+        const token = parts[0];
+        const method = parts[1];
+
+        return fromFetch(`${this.authority}/api/v2.1/logout`, {
+          credentials: 'include',
+          mode: 'cors',
+          method: 'POST',
+          body: null,
+          headers: this.getAuthHeaders(token, method),
+        });
+      }),
+      concatMap(
+        response => {
+          if(response.status === 302) {
+            const redirectUrl = response.headers.get('X-Location') || '';
+            return fromFetch(redirectUrl, {
+              mode: 'cors',
+              method: 'POST',
+              body: null,
```

Comment:
My last commit doesn't seem to remove the kube:admin session cookie from the OAuth server because I'm granted a new token immediately after logging out. The same behaviour occurs whether or not I add `headers: this.getAuthHeaders()` in my request. Any ideas?

Reply:
Was that working previously in this patch set? If so, perhaps you can do a `git bisect` to narrow down which specific change may have broken it.

Reply:
Hmm, [`913a065`](https://github.com/cryostatio/cryostat-web/pull/355/commits/913a065d4bac323ff63518caa2e5888ee95fbaa9) is the first broken commit. In this commit, I piped the `combineLatest` token and method into `setLoggedOut()`, changed `switchMap` to `concatMap`, and removed the auth headers from the redirect url post request. I tried reverting each of these changes individually and the behaviour is still broken. Does anything else look suspicious to you?

Reply:
Strange. What if you check out/build `913a065` and manually test the workflow with your browser's devtools network traffic monitoring tools open? This should show you exactly what requests are made to Cryostat and to OAuth and what parameters and headers were sent, and what the responses were. Once you know what that flow looks like you can check out the previous commit and run the same test again and compare the traffic traces.

It's the `POST` to the Cryostat backend that is meant to delete that `kube:admin` token, right?

Reply:
> It's the POST to the Cryostat backend that is meant to delete that kube:admin token, right?

Yes, that's correct. I'll let you know what I find from the traffic traces soon

Reply:
It turns out that a cookie (presumably the session cookie?) gets sent with the /logout request in the working commit using `credentials: 'include'`. I had removed `credentials: 'include'` in the broken commit because I didn't think I needed to send any additional information. I'll add that line back in